stdlib/buffer.mli

```diff
@@ -131,3 +131,8 @@ val add_channel : t -> in_channel -> int -> unit
 val output_buffer : out_channel -> t -> unit
 (** [output_buffer oc b] writes the current contents of buffer [b]
    on the output channel [oc]. *)
+
+val truncate : t -> int -> unit
+(** [truncate b len] truncates the length of [b] to [len]
+  Note: the intenal byte sequence is not shortenend.
```

Comment:
One thing you can do to avoid existential worries is to add your tests to a file already testing buffer, namely `basic-more/tbuffer.ml`.

In general tests are done by having tests print some output (either by running the program, or the compiler or toplevel on the program), and comparing it to a reference output. The best outputs, I think, are those which give useful information when they fail: `basic-more/tbuffer.ml` just prints the list of test numbers that pass or fail, it's not very useful compared to `lib-printf/pr6534.reference` which gives useful information out failure. But that is a second-order concern: just having a good test that fails at the right time is the important thing here, and being locally consistent with wherever you decide to add your tests suffices. If you don't find a place you like, you can create `tests/lib-buffer`, which would be the natural place for `Buffer` test, and invent your own perfect convention.